### ISSUE TITLE
WS support

### ISSUE THREAD

**Issue by marianozunino** [BackendStack21/fast-gateway#25]
It is possible to foward ws connections?

**Comment by jkyberneees** [comment]
At the moment WebSocket proxying is not supported.

I might support it in future versions if required.

Regards

**Comment by Jamie452** [comment]
Just run into this limitation, would be awesome to have WebSocket support 🙏

**Comment by Kevan0v** [comment]
I need websocket support as well,
Is there any API gateway in Node ready for production (robust, performant) that can integrate easily with service discovery and client-side load-balancing?

**Comment by HoaX7** [comment]
Need ws proxying support asap pls
thanks!

**Comment by jkyberneees** [comment]
Hi community, after a long time, finally I can get back to this feature. 
Could you please comment on this feature proposal:

Usage example:
```js
'use strict'

const gateway = require('fast-gateway')
const PORT = process.env.PORT || 8080

gateway({
  routes: [{
    // ... other HTTP or WebSocket routes
  }, {
    proxyType: 'websocket',
    prefix: '/echo',
    target: 'ws://ws.ifelse.io',
    hooks: {
      onOpen: (ws, searchParams) => {

      }
    }
  }]
}).start(PORT).then(server => {
  console.log(`API Gateway listening on ${PORT} port!`)
})
```

See `WebSocketRoute`:
```ts
 interface WebSocketRoute {
  proxyType: 'websocket';
  proxyConfig?: {};
  prefix: string;
  target: string;
  subProtocols?: []; // https://github.com/faye/faye-websocket-node#subprotocol-negotiation
  hooks?: WebSocketHooks;
}

interface WebSocketHooks {
  onOpen?: Function;
}
```

Regards

**Comment by jkyberneees** [comment]
WebSockets support is now available since v3.1.0!

Regards
